### PR TITLE
feat(aio): display “Searching ..." while building search index

### LINKED ISSUE #15923: [aio] Add a message when the search index is building
**I'm submitting a ...**
```
[x] feature request
```

**Current behavior**
On the new AIO website the search index is built in a webworker in the background when the user access the site for the first time. Until this is done, the search is only partial (the api docs are not available). This is deceptive because the user who gets there for the first time on this browser can think that the search has no results for what he is looking for.

**Expected behavior**
A small "search index is still building" message, with (maybe) a loader would help to indicate that more results are coming.

**Minimal reproduction of the problem with instructions**
Go to https://aio-staging.firebaseapp.com/?search=Date with a new browser and you won't see the "api: DatePipe" result immediately

**What is the motivation / use case for changing the behavior?**
Letting the user know that more results exist

### PR DESCRIPTION
**Pull request by wardbell**
closes #15923

## PR Checklist
Does please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: aio/src/app/search/search-results/search-results.component.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export class SearchResultsComponent implements OnInit, OnDestroy {
 
   private resultsSubscription: Subscription;
   readonly defaultArea = 'other';
+  notFoundMessage = 'Searching ...';
```
Comment (gkalpak) [inline aio/src/app/search/search-results/search-results.component.ts]: Sine we are only showing this initially (and not every time we are searching), we might as well change it to `Building search index...` or something (which is more accurate).

**Review by gkalpak — APPROVED** [review]
LGTM (with one minor suggestion) 👍

**Review by petebacondarwin — APPROVED** [review]

File: aio/src/app/search/search-results/search-results.component.ts (hunk 1 above)
Comment (petebacondarwin) [inline aio/src/app/search/search-results/search-results.component.ts]: Except that the user doesn't really need to know about the internals. As far as they are concerned we are searching for their query.

**Comment by petebacondarwin** [comment]
LGTM as long as Travis is happy

**Comment by sjtrimble** [comment]
Love it! So much better :)

File: aio/src/app/search/search-results/search-results.component.ts (hunk 1 above)
Comment (wardbell) [inline aio/src/app/search/search-results/search-results.component.ts]: More accurate, but an implementation detail. We know what's actually happening. But the reader has no idea what that means. Which is why I'm leaving "as is".

**Pull request merged**